Add simple DeltaF/F transform

jonnew opened the pull request:
- Images are smoothed and numerically integrated to form background.
- Fixed #9

jonnew committed:
Add simple DeltaF/F transform

- Images are smoothed and numerically integrated to form background.

jonnew committed:
Bump version

ChucklesOnGitHub commented:
Consider limiting the Adapatation Rate slider to values >0.
![image](https://github.com/user-attachments/assets/da50cdf4-bc1e-4af3-a860-1866b9681eee)

jonnew committed:
Changes to dff algorithm

- Deal with huge difference in background between focus part of image
  and out of focus

cjsha commented on OpenEphys.Miniscope/DeltaFOverF.cs:
```diff
@@ -0,0 +1,92 @@
+
+namespace OpenEphys.Miniscope
+{
+    [Description("Performs a naïve, causal deltaF/F calculation on a series of images. In general, regardless of how its computed, deltaF/F should not be used for quantitative analysis due to to its dependence on accurate background estimation and the (generally) nonlinear relationship between fluorescence intensity and the underlying independent variable of interest (e.g. Calcium levels).")]
```
I'm hesitant about the severity in which DF/F is described here.

jonnew replied:
I toned it down and provided more context.

cjsha reviewed (approved):
I left some feedback, but I'm going to approve so you can merge after the feedback is addressed or discarded.

Tangentially related comment: I've been wondering if it's possible to separate background light and the cells' baseline florescence. In other words, is it possible to subtract all the light that the miniscope would see if there were no cells in its focal plane (leaked excitation light, out-of-focus emission light, etc.). Then the DF/F processing can be done on those background-subtracted images. The difficulty is figuring out what light is actually background and not fluorescence that we do want to measure.

jonnew commented on OpenEphys.Miniscope/DeltaFOverF.cs:
```diff
@@ -0,0 +1,58 @@
+                    var sigma = Sigma % 2 == 0 ? Sigma + 1 : Sigma;
+
+                    CV.Convert(input, image); // Convert to floating point type
+                    CV.Smooth(image, image, SmoothMethod.Gaussian, 2*sigma + 1, 0, sigma);
```
Im trying to keep this simple, so probably no.

jonnew committed:
Address review comments
